src/mountinfo.rs

```diff
@@ -0,0 +1,418 @@
+use std::fmt;
+use std::ffi::{OsStr, OsString};
+use std::os::unix::ffi::{OsStrExt, OsStringExt};
+use std::path::PathBuf;
+use std::borrow::Cow;
+use std::error::Error;
+
+use libc::c_ulong;
+use libc::{MS_RDONLY, MS_NOSUID, MS_NODEV, MS_NOEXEC, MS_SYNCHRONOUS};
+use libc::{MS_MANDLOCK, MS_DIRSYNC, MS_NOATIME, MS_NODIRATIME};
+use libc::{MS_RELATIME, MS_STRICTATIME};
+
+#[derive(Debug)]
+pub struct ParseError(String, usize, String);
```

Comment:
It's hard to reason what these fields mean. At least make it a struct with named fields. Better make a quick_error with contexts.